### DISCUSSION TITLE
How are taskLocationRoot and taskTagRoot able to display a dropdown on promoted location/tag labels?

### DISCUSSION THREAD

**Discussion by laundmo** [TriliumNext/Trilium#3449] [Q&A] [answered]
Best i can tell, this is what happens:
- A root note with `taskLocationRoot`
- Children of it have `taskLocationNote=name`
- Any other note can have a `#label:location=promoted,single,text`
- \*\*magic happens\*\*
- I can now get a dropdown of all `taskLocationNote` names in that promoted `location` label

I can't find anything about how this works. I'd love to be able to define my own categories of label options to choose from, but without knowing where and how the magic happens to make this work i'm really confused.

I've looked at the scripts defined in Task Manager and they only seem to run *after* a attribute is set, not beforehand to offer the dropdown. So, how does it work? what do i need to do to create my own categories from which one can pick (in my case i would want `projectId` and `person` categories to use in promoted attributes and scripts)

**Comment by laundmo** [comment, marked as answer]
I figured it out:
The dropdown is NOT filled by getting the `taskLocationNote=name` from existing location notes, the location notes are added by writing down a location in the `location` label. The dropdown is just built from other notes using the same label. Now the script makes a lot more sense too.

**Reply by jinxiaoman** [reply]
hi. I have the same problem as you, enen. Didn't understand it, is there an example?
